parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestDictionaryReader.java

```diff
@@ -0,0 +1,206 @@
+              .append("binary_field", b)
+              .append("int32_field", intValues[index])
+              .append("int64_field", longValues[index])
+              .append("double_field", intValues[index] * 1.0)
+              .append("float_field", ((float) (intValues[index] * 2.0)));
+
+      writer.write(group);
+    }
+    writer.close();
+  }
+
+  @BeforeClass
+  public static void prepareFile() throws IOException {
+    cleanup();
+
+    boolean dictionaryEnabled = true;
+    boolean validating = false;
+    GroupWriteSupport.setSchema(schema, conf);
+    SimpleGroupFactory f = new SimpleGroupFactory(schema);
+    ParquetWriter<Group> writer = new ParquetWriter<Group>(
+            file,
+            new GroupWriteSupport(),
+            GZIP, 1024*1024, 1024, 1024*1024,
+            dictionaryEnabled, validating, PARQUET_1_0, conf);
+    writeData(f, writer);
+  }
+
+  @AfterClass
+  public static void cleanup() throws IOException {
+    FileSystem fs = file.getFileSystem(conf);
+    if (fs.exists(file)) {
+      fs.delete(file, true);
+    }
+  }
+
+  @Test
+  public void testBinaryRead() throws Exception {
+    MessageType requestedSchema = parseMessageType("message test { required binary binary_field;}");
+    ParquetMetadata metadata = ParquetFileReader.readFooter(conf, file);
+    ParquetFileReader fileReader = new ParquetFileReader(conf, metadata.getFileMetaData(), file, metadata.getBlocks(), requestedSchema.getColumns());
+    DictionaryPageReadStore dictionaryReadStore = fileReader.readNextRowGroupDictionaries(requestedSchema);
+    ColumnDescriptor column = requestedSchema.getColumns().get(0);
+    DictionaryPageReader dictionaryPageReader = dictionaryReadStore.getDictionaryPageReader(column);
+    DictionaryPage dictionaryPage = dictionaryPageReader.readDictionaryPage();
+    Dictionary dictionary = dictionaryPage.getEncoding().initDictionary(column, dictionaryPage);
+
+    for (int i = 0; i < intValues.length; i++) {
+      assertEquals(String.valueOf((char) ((i % intValues.length) + 'a')), dictionary.decodeToBinary(i).toStringUsingUTF8());
+    }
+    fileReader.close();
+  }
+
+  @Test
+  public void testIntRead() throws Exception {
+    MessageType requestedSchema = parseMessageType("message test { required int32 int32_field;}");
+    ParquetMetadata metadata = ParquetFileReader.readFooter(conf, file);
+    ParquetFileReader fileReader = new ParquetFileReader(conf, metadata.getFileMetaData(), file, metadata.getBlocks(), requestedSchema.getColumns());
+    DictionaryPageReadStore dictionaryReadStore = fileReader.readNextRowGroupDictionaries(requestedSchema);
+    ColumnDescriptor column = requestedSchema.getColumns().get(0);
+    DictionaryPageReader dictionaryPageReader = dictionaryReadStore.getDictionaryPageReader(column);
+    DictionaryPage dictionaryPage = dictionaryPageReader.readDictionaryPage();
+    Dictionary dictionary = dictionaryPage.getEncoding().initDictionary(column, dictionaryPage);
```

Comment:
You should wrap this logic in a class that returns Dictionary objects.
Typically user code does not want to interact with Pages but the final decoded data.